### PR TITLE
Event docs, typings fixes

### PR DESCRIPTION
**Pull request by bsian03**
For update events, I have removed props that only appear in REST

### CODE REVIEW

File: lib/gateway/Shard.js [hunk 1]
```diff
@@ -1004,11 +1006,15 @@ class Shard extends EventEmitter {
                         }
                     }
                 }
+                if(packet.d.emoji.animated === undefined) {
+                    packet.d.emoji.animated = false;
+                }
```
Comment (bsian03) [inline lib/gateway/Shard.js]:
```suggestion
```

File: lib/gateway/Shard.js [hunk 2]
```diff
@@ -1004,11 +1006,15 @@ class Shard extends EventEmitter {
...
                 /**
                 * Fired when someone removes a reaction from a message
                 * @event Client#messageReactionRemove
                 * @prop {Message | Object} message The message object. If the message is not cached, this will be an object with `id`, `channel`, and if inside a guild, `guildID` keys. If the channel is not cached, channel key will be an object with only an id. `guildID` will be present if the message was sent in a guild channel. No other property is guaranteed
                 * @prop {Object} emoji The reaction emoji object
+                * @prop {Boolean} emoji.animated Whether the emoji is animated or not
```
Comment (bsian03) [inline lib/gateway/Shard.js]:
```suggestion
                * @prop {Boolean?} emoji.animated Whether the emoji is animated or not
```

File: lib/gateway/Shard.js [hunk 3]
```diff
@@ -1067,11 +1073,15 @@ class Shard extends EventEmitter {
                         }
                     }
                 }
+                if(packet.d.emoji.animated === undefined) {
+                    packet.d.emoji.animated = false;
+                }
```
Comment (bsian03) [inline lib/gateway/Shard.js]:
```suggestion
```

File: lib/gateway/Shard.js [hunk 4]
```diff
@@ -1067,11 +1073,15 @@ class Shard extends EventEmitter {
...
                 /**
                 * Fired when someone removes all reactions from a message for a single emoji
                 * @event Client#messageReactionRemoveEmoji
                 * @prop {Message | Object} message The message object. If the message is not cached, this will be an object with `id` and `channel` keys. If the channel is not cached, channel key will be an object with only an id. No other property is guaranteed
                 * @prop {Object} emoji The reaction emoji object
+                * @prop {Boolean} emoji.animated Whether the emoji is animated or not
```
Comment (bsian03) [inline lib/gateway/Shard.js]:
```suggestion
                * @prop {Boolean?} emoji.animated Whether the emoji is animated or not
```

**Review by bsian03 — COMMENTED** [review]
Backwards GitHub app ftw

**Review by abalabahaha — APPROVED** [review]

**Pull request merged**